docs/decisions/0004-downstream-effect-events.rst

```diff
@@ -0,0 +1,105 @@
+Implementation of Event Driven Architecture for Exam Downstream Effects
+=======================================================================
+
+Status
+------
+
+Draft (circa Sept 2023)
+
+Context
+-------
+We are porting any downstream effects of exam submission and review from the legacy exams system (edx-proctoring) to this one (edx-exams).
+Most of these downstream effects will reflect "calls" that edx-proctoring makes to various edx-platform services,
+such as grades, credits, and certificates. Note that these "calls" are not network requests, but calls to functions in edx-platform,
+which edx-proctoring can call directly since it is a plugin that is installed directly into edx-platform.
+
+Decision
+--------
+We've decided to use an event driven architecture because unlike the legacy system (edx-porctoring), edx-exams is an independent service.
+As such, we plan to use the event bus to send info to edx-platform services without needing a response as one would in a REST framework.
+
+Below are lists of downstream effects of exam submission and review that we will or will not be translating into events in edx-exams as part of this decision.
+
+Downstream effects to be implemented as events in the short term:
+*****************************************************************
+
+  * Grades Override - A python API call to the grades app to generate a grade override when an exam attempt is rejected.
+
+  * Undo Grades Override - A python API call to the grades app to delete a grade override when marking a rejected attempt as verified.
+
+  * Instructor Delete Attempt - A python API call to the instructor app to delete an exam attempt.
+
+  * Instructor Complete Attempt - A python API call to the instructor app to mark an exam subsection as completed.
+
+  * Invalidate Certificate - A python API call to the certificates app to invalidate a leaner’s edx certificate for a course.
+
+  * Set Credit Requirement Status - A python API call to the credits app to create or modify a learner’s credit requirement status for an exam.
+
+Downstream effects that will be implemented in the long term:
```

Comment:
You should be explicit that "in the long term" means, they are not implemented as part of accepting this decision. Things changes in the future. I just don't want the record read as a commitment to follow up our long term plans.